src/test/java/io/gaia_app/credentials/CredentialsServiceIT.kt

```diff
@@ -0,0 +1,73 @@
+package io.gaia_app.credentials
+
+import io.gaia_app.encryption.EncryptionService
+import io.gaia_app.vault.VaultConfiguration
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.client.AutoConfigureWebClient
+import org.springframework.boot.test.autoconfigure.web.client.RestClientTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.http.MediaType
+import org.springframework.test.util.ReflectionTestUtils
+import org.springframework.test.web.client.MockRestServiceServer
+import org.springframework.test.web.client.match.MockRestRequestMatchers.header
+import org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo
+import org.springframework.test.web.client.response.MockRestResponseCreators
+import org.springframework.vault.core.VaultTemplate
+import org.springframework.web.client.RestTemplate
+
+@RestClientTest(components = [CredentialsService::class, VaultConfiguration::class],
+properties = ["gaia.vault.enabled=true",
+"gaia.vault.authentication.token=vault-token",
+"gaia.vault.uri=http://vault-host:8200"
+])
+@AutoConfigureWebClient(registerRestTemplate = true)
+class CredentialsServiceIT {
+
+    @Autowired
+    lateinit var credentialsService: CredentialsService
+
+    @MockBean
+    lateinit var credentialsRepository: CredentialsRepository
+
+    @MockBean
+    lateinit var encryptionService: EncryptionService
+
+    @Autowired
+    lateinit var vaultTemplate: VaultTemplate
+
+    @Test
+    fun `loadAWSCredentialsFromVault should call vault aws secret engine`(){
+        // given
+        val jsonResponse = """{
```

Comment:
we can clean the format here, no ?

Reply:
do you mean having a better indentation ? yes